From: Peter Lemenkov <lemenkov@gmail.com>
Date: Wed, 31 Dec 2025 00:49:24 +0100
Subject: [PATCH] Raise minimal OpenSSL version to 1.1.1

This version has TLS 1.3 support and this is a first version where it is
possible to dump crypto keys.

Signed-off-by: Peter Lemenkov <lemenkov@gmail.com>

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 141de0c..1369a55 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -134,7 +134,7 @@ if(USE_WOLFSSL)
     endif()
     add_definitions("-DUSE_WOLFSSL" "-DOPENSSL_ALL")
 else()
-    pkg_search_module(SSL REQUIRED openssl>=1.1.0)
+    pkg_search_module(SSL REQUIRED openssl>=1.1.1)
     add_definitions("-DUSE_OPENSSL")
     if(TLS_KEY_LOGGING)
         add_definitions("-DUSE_OPENSSL_KL")
diff --git a/docs/installation.rst b/docs/installation.rst
index 92b494b..641a59b 100644
--- a/docs/installation.rst
+++ b/docs/installation.rst
@@ -63,7 +63,7 @@ Installing SIPp
 
     + C++ Compiler
     + curses or ncurses library
-    + OpenSSL >= 1.1.0 or WolfSSL >= 3.15.0
+    + OpenSSL >= 1.1.1 or WolfSSL >= 3.15.0
     + For pcap play support: libpcap and libnet
     + For SCTP support: lksctp-tools
     + For distributed pauses: `Gnu Scientific Libraries`_
diff --git a/src/sslsocket.cpp b/src/sslsocket.cpp
index 0f91622..00917d5 100644
--- a/src/sslsocket.cpp
+++ b/src/sslsocket.cpp
@@ -23,8 +23,7 @@
 #include "sipp.hpp"
 #include "sslsocket.hpp"
 
-#if (USE_OPENSSL_KL && OPENSSL_VERSION_NUMBER >= 0x10101000L)
-#define HAVE_KEYLOG_CB
+#if USE_OPENSSL_KL
 #define KEYLOG_ENV "SSLKEYLOGFILE"
 #endif
 
@@ -99,7 +98,7 @@ static int passwd_call_back_routine(char *buf, int size, int /*flag*/, void *pas
     return(strlen(buf));
 }
 
-#ifdef HAVE_KEYLOG_CB
+#ifdef USE_OPENSSL_KL
 static const char *keylog_file()
 {
     char *kl_env = getenv(KEYLOG_ENV);
@@ -148,7 +147,7 @@ static void sip_tls_keylog_callback(const SSL *ssl, const char *line)
         write_keylog_line(keylog_file_name, line);
     }
 }
-#endif // HAVE_KEYLOG_CB
+#endif // USE_OPENSSL_KL
 
 /****** SSL error handling *************/
 const char *SSL_error_string(int ssl_error, int orig_ret)
@@ -453,12 +452,12 @@ enum tls_init_status TLS_init_context(void)
         return TLS_INIT_ERROR;
     }
 
-#ifdef HAVE_KEYLOG_CB
+#ifdef USE_OPENSSL_KL
     if (keylog_file()) {
         SSL_CTX_set_keylog_callback(sip_trp_ssl_ctx_client, sip_tls_keylog_callback);
         SSL_CTX_set_keylog_callback(sip_trp_ssl_ctx, sip_tls_keylog_callback);
     }
-#endif // HAVE_KEYLOG_CB
+#endif // USE_OPENSSL_KL
 
 
     return TLS_INIT_NORMAL;
